patches/server/0804-More-Teleport-API.patch

```diff
@@ -0,0 +1,138 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Owen1212055 <23108066+Owen1212055@users.noreply.github.com>
+Date: Sun, 5 Sep 2021 12:15:59 -0400
+Subject: [PATCH] More Teleport API
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
+index 0e1dcfdf846f978340380352e092375ffa089cb0..f0c0939d2896c2e0f8282d68a02d06682dee0bc3 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
+@@ -555,17 +555,39 @@ public abstract class CraftEntity implements org.bukkit.entity.Entity {
+         return this.teleport(location, TeleportCause.PLUGIN);
+     }
+ 
++    // Paper start - Teleport passenger API
+     @Override
+-    public boolean teleport(Location location, TeleportCause cause) {
++    public boolean teleport(@org.jetbrains.annotations.NotNull Location location, @org.jetbrains.annotations.NotNull TeleportCause cause) {
++        return teleport(location, cause, false);
++    }
++    @Override
++    public boolean teleport(@org.jetbrains.annotations.NotNull Location location, boolean keepPassengers) {
++        return teleport(location, TeleportCause.PLUGIN, keepPassengers);
++    }
++
++    @Override
++    public boolean teleport(@org.jetbrains.annotations.NotNull Location location, boolean keepPassengers, boolean dismount) {
++        return teleport(location, TeleportCause.PLUGIN, keepPassengers, dismount);
++    }
++
++    @Override
++    public boolean teleport(@org.jetbrains.annotations.NotNull Location location, @org.jetbrains.annotations.NotNull TeleportCause cause, boolean keepPassengers) {
++        return teleport(location, keepPassengers, true);
++    }
```

Comment:
I was looking before and generally I never saw default methods in interfaces, only a couple places which some had a // TODO: Move to implementation 

Is there a reason why they aren’t used often? I can still do it but I’m curious.

Reply:
Probably because Spigot is scared of using anything new, including Java features, and because people tried keeping things similar to it - which is not a goal we should follow if it goes against simplicity/readability. Default methods definitely are used a good bunch in Paper API as well